### PR TITLE
DEV: Drop unused column email_tokens.token

### PR DESCRIPTION
**Pull request by udan11**
email_tokens.token_hash is used instead.

### CODE REVIEW

**Review by eviltrout — APPROVED** [review]

**Comment by eviltrout** [comment]
There's a failing test you might need to fix.

File: db/migrate/20211206160211_create_index_on_email_tokens_token_hash.rb
Comment (CvX) [inline db/migrate/20211206160211_create_index_on_email_tokens_token_hash.rb:3]: This one will be merged in https://github.com/discourse/discourse/pull/15202, right?

Comment (udan11) [reply db/migrate/20211206160211_create_index_on_email_tokens_token_hash.rb]: That is correct. This migration has been already merged in another PR and this commit is building on top of that.

**Comment by SamSaffron** [comment]
this is weird, I thought we require 2 reviews for PRs including migrations @CvX ?

File: db/post_migrate/20211206160212_drop_token_from_email_tokens.rb
Comment (SamSaffron) [inline db/post_migrate/20211206160212_drop_token_from_email_tokens.rb:11]: this indirection and constant feels a bit weird. Simply: 

```
Migration::ColumnDropper.execute_drop('email_tokens', ['token'])
```

?

**Review by SamSaffron — APPROVED** [review]

File: db/post_migrate/20211206160212_drop_token_from_email_tokens.rb
Comment (gschlager) [reply db/post_migrate/20211206160212_drop_token_from_email_tokens.rb:11]: Unfortunately we need that constant, otherwise restoring a backup might fail due to missing functions in the `discourse_functions` schema.

https://github.com/discourse/discourse/blob/d2c5165052fd8ef588bb16523d616bb2732ec379/lib/backup_restore/database_restorer.rb#L156-L180

**Comment by CvX** [comment]
> this is weird, I thought we require 2 reviews for PRs including migrations @CvX ?

@sam, are you asking about GitHub repo settings or about our policy?

In terms of settings, we require one approval before a PR can be merged. I don't think there is a way* to make that number depend on arbitrary factors (e.g. there being changes with a particular path prefix, like `db/migrate`) but I'm basing this only on reading the docs, since I don't see our settings. (that's a GOO-exclusive thing)

And the policy - I can't find it being actually codified anywhere. I'm sure it has been mentioned in one of the topics on branch protections, reviews, or SOC2, but I'm not sure it was included in any of the OPs in #policy or #dev. 🤔 

---

* just in theory, we could make an Action (or maybe there is one already) that checks if there are changes to a given path, and uses GH API to check the number of approvals. But that require us to enable the setting that prevents merging when there are failed checks. And we don't want to do that because we want to be able to merge stuff even when there's a flaky test, or when we just update a PR with a comment typo-fix and don't want to wait another 20 minutes.

**Pull request merged**
